controllers/submariner/metrics_proxy_resources.go

```diff
@@ -0,0 +1,107 @@
+	"github.com/submariner-io/submariner-operator/pkg/names"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// nolint:wrapcheck // No need to wrap errors here.
+func (r *Reconciler) reconcileMetricsProxyDaemonSet(instance *v1alpha1.Submariner, reqLogger logr.Logger) (*appsv1.DaemonSet,
+	error,
+) {
+	return helpers.ReconcileDaemonSet(instance, newMetricsProxyDaemonSet(instance, names.MetricsProxyComponent), reqLogger,
+		r.config.ScopedClient, r.config.Scheme)
+}
+
+func newMetricsProxyDaemonSet(cr *v1alpha1.Submariner, name string) *appsv1.DaemonSet {
+	labels := map[string]string{
+		"app":       name,
+		"component": "metrics",
+	}
+
+	daemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      name,
+			Labels:    labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
+				"app": name,
+			}},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "gateway-metrics-proxy",
+							Image:           getImagePath(cr, names.MetricsProxyImage, names.MetricsProxyComponent),
+							ImagePullPolicy: helpers.GetPullPolicy(cr.Spec.Version, cr.Spec.ImageOverrides[names.MetricsProxyComponent]),
+							Env: []corev1.EnvVar{
+								{Name: "NODE_IP", ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "status.hostIP",
+									},
+								}},
+							},
+							Command: []string{"/usr/bin/nc"},
+							Args:    []string{"-v", "-lk", "-p", fmt.Sprint(gatewayMetricsServerPort), "-e", "nc", "$(NODE_IP)", constants.MetricsGWPort},
+						},
+					},
+					NodeSelector: map[string]string{"submariner.io/gateway": "true"},
+					// The MetricsProxy Pod must be able to run on any flagged node, regardless of existing taints
+					Tolerations: []corev1.Toleration{{Operator: corev1.TolerationOpExists}},
+				},
+			},
+		},
+	}
+
+	if cr.Spec.GlobalCIDR != "" {
+		gnContainer := corev1.Container{
+			Name:            "globalnet-metrics-proxy",
+			Image:           getImagePath(cr, names.MetricsProxyImage, names.MetricsProxyComponent),
+			ImagePullPolicy: helpers.GetPullPolicy(cr.Spec.Version, cr.Spec.ImageOverrides[names.MetricsProxyComponent]),
+			Env: []corev1.EnvVar{
+				{Name: "NODE_IP", ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "status.hostIP",
+					},
+				}},
+			},
+			Command: []string{"/usr/bin/nc"},
+			Args:    []string{"-v", "-lk", "-p", fmt.Sprint(globalnetMetricsServerPort), "-e", "nc", "$(NODE_IP)", constants.MetricsGNPort},
+		}
+
+		daemonSet.Spec.Template.Spec.Containers = append(daemonSet.Spec.Template.Spec.Containers, gnContainer)
+	}
+
+	return daemonSet
+}
```

Comment:
```suggestion
}

func metricProxyContainer(cr *v1alpha1.Submariner, name string, hostPort, podPort int) *corev1.Container {
	return &corev1.Container{
		Name:            name,
		Image:           getImagePath(cr, names.MetricsProxyImage, names.MetricsProxyComponent),
		ImagePullPolicy: helpers.GetPullPolicy(cr.Spec.Version, cr.Spec.ImageOverrides[names.MetricsProxyComponent]),
		Env: []corev1.EnvVar{
			{Name: "NODE_IP", ValueFrom: &corev1.EnvVarSource{
				FieldRef: &corev1.ObjectFieldSelector{
					FieldPath: "status.hostIP",
				},
			}},
		},
		Command: []string{"/usr/bin/nc"},
		Args:    []string{"-v", "-lk", "-p", fmt.Sprint(hostPort), "-e", "nc", "$(NODE_IP)", fmt.Sprint(podPort)},
	}
}
```

Generally I think it's a good idea to reuse code if possible, and here it's quite easy (instead of maintaining a duplicate).

Not crucial, but I do find such code easier to maintain.